### PR TITLE
feat: add catalog screen

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#82772

### CODE REVIEW

File: packages/apps/sale/src/screens/cart/index.ts
Comment (vhu-axelor) [inline packages/apps/sale/src/screens/cart/index.ts:1]: Export CatalogScreen and CatalogVariantScreen

File: packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx
Comment (vhu-axelor) [inline packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx:1]: On the design models the plus button is above

File: packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx [hunk 1]
```diff
@@ -16,97 +16,140 @@
...
+        product={_product}
+        qty={cartLine?.qty}
+        unit={cartLine?.unit?.name}
+        hideBadgeInformation={hideBadgeInformation}
+        onPress={() => {
+          !product == null &&
+            navigation.navigate('CartLineDetailsScreen', {
+              cartLineId: cartLine?.id,
+            });
+        }}
```
Comment (vhu-axelor) [inline packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx:152]: Navigation doesn't work anywhere

File: packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx [hunk 2]
```diff
@@ -16,111 +16,139 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, {useCallback, useMemo} from 'react';
```
Comment (lme-axelor) [inline packages/apps/sale/src/components/molecules/CartLineActionCard/CartLineActionCard.tsx:19]: Create a new card to manage the catalog view, there are too much differences between the two

File: packages/apps/sale/src/screens/cart/CatalogScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+const CatalogScreen = ({}) => {
+  const dispatch = useDispatch();
+  const I18n = useTranslator();
+
+  const {userId} = useSelector((state: any) => state.auth);
+  const {productList, moreLoading, isListEnd, loadingList} = useSelector(
+    (state: any) => state.sale_product,
+  );
+
+  const {activeCart} = useSelector((state: any) => state.sale_cart);
```
Comment (lme-axelor) [inline packages/apps/sale/src/screens/cart/CatalogScreen.tsx:43]: Move this between the two existing selectors

File: packages/apps/sale/src/screens/cart/CatalogVariantScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useEffect, useMemo} from 'react';
```
Comment (lme-axelor) [inline packages/apps/sale/src/screens/cart/CatalogVariantScreen.tsx:19]: Remove the variant catalog screen please, we found a simplier solution for this. We will do it later

File: packages/apps/sale/src/screens/cart/CatalogVariantScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  displayItemName,
+  useDispatch,
+  useSelector,
+  useTranslator,
+} from '@axelor/aos-mobile-core';
+import {fetchVariantProduct} from '../../features/productSlice';
+import {CartLineActionCard} from '../../components';
+import {fetchActiveCart} from '../../features/cartSlice';
+
+const catalogScanKey = 'sale_catalog_scanKey';
```
Comment (lme-axelor) [inline packages/apps/sale/src/screens/cart/CatalogVariantScreen.tsx:32]: Just a note for later as this screen will disappear, two screens should not have the same scan key

**Pull request merged**